testerror/error.go

```diff
@@ -1,6 +1,8 @@
 // Licensed under the MIT license, see LICENCE file for details.
 
-package quicktest
+// Package testerror provides error types which can be used when implementing
+// checkers.
+package testerror
```

Comment:
I don't think that the testerror package pulls its weight. I'd leave the BadCheck stuff in quicktest and leave mismatch and friends unexported (it's not that clear that it would be useful to external check authors anyway).
tbh I might be inclined just to use:

   var ErrBadCheck = errgo.New("bad test check")

then you could do

    if errgo.Cause(err) == ErrBadCheck

and dispense with BadCheckf and IsBadCheck.

Reply:
I put back the error as it was originally.